jerry-core/jerryscript.h

```diff
@@ -166,6 +166,29 @@ typedef void (*jerry_object_free_callback_t) (const uintptr_t native_p);
 typedef bool (*jerry_object_property_foreach_t) (const jerry_value_t property_name,
                                                  const jerry_value_t property_value,
                                                  void *user_data_p);
+/**
+ * Type information of a native handle.
+ */
+typedef struct
+{
+  jerry_object_free_callback_t free_cb; /**< the free callback of the native handle */
+} jerry_native_handle_info_t;
+
+/**
+ * The name of the jerry_native_handle_info_t struct.
+ */
+#define JERRY_NATIVE_HANDLE_INFO_FOR_CTYPE(c_type) _jerry_native_handle_info_##c_type
+
+/**
+ * Define a native handle typeinfo based on the C type and free callback.
+ */
+#define JERRY_DEFINE_NATIVE_HANDLE_INFO(c_type, native_free_cb) \
+  static const jerry_native_handle_info_t JERRY_NATIVE_HANDLE_INFO_FOR_CTYPE (c_type) = \
+  { \
+    .free_cb = (jerry_object_free_callback_t) native_free_cb \
+  }
```

Comment:
Check Linux kernel, that is very common there.

Personally I don't like macros in public headers which hides basic operations. So hard to understand the code later. I want to see `static const jerry_native_handle_info_t` because its purpose is clear rather than `JERRY_DEFINE_NATIVE_HANDLE_INFO()`, which forces you to do extra searches in the code.